AlphaWallet/Market/Coordinators/UniversalLinkCoordinator.swift

```diff
@@ -27,42 +29,53 @@ enum MagicLinkURL {
 
     case eip681(URL)
     case walletConnect(url: AlphaWallet.WalletConnect.ConnectionUrl, source: WalletConnectSource)
+    case specialUrl(RPCServer, URL)
 
     init?(url: URL) {
-        if let eip681Url = Self.hasEip681Path(in: url) {
+        if let eip681Url = Self.functional.hasEip681Path(in: url) {
             self  = .eip681(eip681Url)
         } else if let (wcUrl, source) = Self.functional.hasWalletConnectPath(in: url) {
             self = .walletConnect(url: wcUrl, source: source)
+        } else if let (server, url) = Self.functional.hasSpecialPath(in: url) {
+            self = .specialUrl(server, url)
         } else {
             return nil
         }
     }
+}
 
+extension MagicLinkUrl {
+    class functional {}
+}
+
+extension MagicLinkUrl.functional {
     //E.g. https://aw.app/ethereum:0x89205a3a3b2a69de6dbf7f01ed13b2108b2c43e7/transfer?address=0x8e23ee67d1332ad560396262c48ffbb01f93d052&uint256=1
-    private static func hasEip681Path(in url: URL) -> URL? {
-        guard let magicLink = RPCServer(withMagicLink: url), url.path.starts(with: Self.eip681Path) else { return nil }
+    static func hasEip681Path(in url: URL) -> URL? {
+        guard let magicLink = RPCServer(withMagicLink: url), url.path.starts(with: MagicLinkUrl.eip681Path) else { return nil }
```

Comment:
This one's old but since we are modifying it, can you rename `magicLink` to `server` or `magicLinkServer`?